### PR TITLE
do not check for nullness

### PR DESCRIPTION
**Pull request by pxpm**
This is a fix I am already implementing in #3410 
image

But since #3410 could take a little longer to be merged, I created this separate PR to advance this fix in time. 

Thanks,
Pedro

### CODE REVIEW

**Comment by tabacitu** [comment]
You've been on fire this week @pxpm ! Great work! 🔥🔥🔥

**Review by promatik — APPROVED** [review]
Tested, it's working 👌

**Comment by kitzberger** [comment]
Not sure if this is related to this patch: but after upgrading to 4.1.44 we cannot **update** our data model any longer.

Our scenario is:
* user_id is not nullable in the DB
* user_id is a hidden field in the create form and only serverside set via the creating event to `backpack_user()->id`
* user_id is not present in the update form at all

With 4.1.44 there's an exception when trying to update the model in the DB:
Bildschirmfoto vom 2021-05-13 10-00-21

**Comment by tabacitu** [comment]
Ouch! Sorry to hear that @kitzberger . 

I've opened https://github.com/Laravel-Backpack/CRUD/issues/3713 to discover what's happened, let's move the conversation there please.

**Comment by tabacitu** [comment]
@pxpm I think the easiest way to fix https://github.com/Laravel-Backpack/CRUD/issues/3713 ASAP is to revert this PR. Then we can figure out a workaround or better solution, most likely by working more on https://github.com/Laravel-Backpack/CRUD/pull/3747

I'm inclined to revert this PR for the time being. Can you please ELI5 
- why this PR was here in the first place? What problem it was fixing?
- what will happen if we revert this PR?

Thanks!

**Comment by pxpm** [comment]
@tabacitu 

without this fix trying to save NULL in a database for a relation-field would not work because we would only enter the conditional if `is not null`. 

This is the referenced issue: https://github.com/Laravel-Backpack/CRUD/issues/3646

I've had encountered this problem too while working in #3410 so I had already removed this condition check from there too.

Please if you are unsure about the fix proposed in #3747 just revert this PR and I will try to figure another way of fixing the described issue. 

Best,
Pedro
